test/test_local_io.py

```diff
@@ -640,6 +664,70 @@ def test_decompressor_iterdatapipe(self):
         with self.assertRaisesRegex(TypeError, "has no len"):
             len(tar_decompress_dp)
 
+    def test_zlibfile_readall(self):
+        uncompressed_data_test_cases = [b"", b"some data", 10_000 * b"some data"]
+        for uncompressed_data in uncompressed_data_test_cases:
```

Comment:
I didn't find an obvious way of using pytest parameterized tests that was compatible with `expecttest`, so I resorted to just manually iterating over a list of inputs.